src/iuse.cpp

```diff
@@ -1901,63 +1902,108 @@ int iuse::mycus(player *p, item *it, bool t, const tripoint &pos)
     return it->type->charges_to_use();
 }
 
-// TOOLS below this point!
+// Types of petfood for each different case
+enum Petfood
+{
+    DOGFOOD,
+    CATFOOD,
+    CATTLEFODDER
+};
 
-int petfood(player *p, item *it, bool is_dogfood)
+int petfood( player *p, item *it, Petfood animal_food_type )
 {
     tripoint dirp;
-    if (!choose_adjacent(string_format(_("Put the %s where?"), it->tname().c_str()), dirp)) {
+    if( !choose_adjacent( string_format( _( "Put the %s where?" ), it->tname().c_str() ), dirp ) ) {
         return 0;
     }
     p->moves -= 15;
     const int mon_idx = g->mon_at( dirp, true );
-    if(mon_idx != -1) {
-        monster &mon = g->zombie( mon_idx );
-        if(mon.type->id == (is_dogfood ? mon_dog : mon_cat)) {
-            p->add_msg_if_player(m_good, is_dogfood
-                ? _("The dog seems to like you!")
-                : _("The cat seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with cats."));
-            mon.friendly = -1;
-            if( is_dogfood ) {
-                mon.add_effect( effect_pet, 1, num_bp, true );
-            }
-        } else if( is_dogfood && mon.type->id == mon_dog_thing ) {
-            p->deal_damage( &mon, bp_hand_r, damage_instance( DT_CUT, rng( 1, 10 ) ) );
-            p->add_msg_if_player( m_bad, _( "You want to feed it the dog food, but it bites your fingers!" ) );
-            if( one_in( 5 ) ) {
-                p->add_msg_if_player( _( "Apparently it's more interested in your flesh than the dog food in your hand!" ) );
-            }
+
+    if( npc *const person_ = g->critter_at<npc>( dirp ) ) {
+        npc &person = *person_;
+        query_yn( _( "Are you sure you want to feed a person the dog food?" ) );
```

Comment:
After a2ed466ab7dc8f95347073380019620d4f9b2075 when I compile and test it with NPCs, I get that query when applying dog food to NPCs.
 Unless I'm misunderstanding what you mean by "hung in the air".